src/Sentry/Protocol/Exceptions/SentryStackFrame.cs

```diff
@@ -248,6 +257,13 @@ public void WriteTo(Utf8JsonWriter writer)
                 writer.WriteNumber("symbol_addr", symbolAddress);
             }
 
+            // Instruction address
+            if (InstructionAddress is { } instructionAddress &&
+                Regex.IsMatch(instructionAddress, @"\A\b0[xX][0-9a-fA-F]+\b\Z"))
+            {
+                writer.WriteString("instruction_addr", instructionAddress.ToLower());
+            }
```

Comment:
I'd rather we dont' have such logic buried in the serialization code.

We can add that elsewhere, or nowhere really (let it hit the server and the error there will alert the user it was malformed. That will only affect users creating frames "by hand" and doing it wrong, not us. (Assuming we get it right :) )

Also, why `ToLower`? it'll allocate a string. Can we just make sure we get the value in its right format into the object?

```suggestion
            if (!string.IsNullOrWhiteSpace(instructionAddress))
            {
                writer.WriteString("instruction_addr", instructionAddress);
            }
```

Reply:
Alright, got rid of that code :D